### PR TITLE
pot: init at 0.5.0

### PR DESCRIPTION
**Pull request by linsui**
###### Description of changes

A translation app.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 232836` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pot</li>
  </ul>
</details>

**Comment by Janik-Haag** [comment]
Also you seem to be missing a desktopItem, for examples you can grep through nixpkgs.

**Comment by linsui** [comment]
It seems that the tauri cli can generate that. I'll check that, thanks!

File: pkgs/applications/misc/pot/default.nix [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    libayatana-appindicator
+    xdotool
+  ];
+
+  preBuild = ''
+    cd ..
+    export HOME=$(mktemp -d)
+    pnpm config set store-dir ${pnpm-deps}
+    pnpm install --offline
+    pnpm build
```
[the remark this answers is not in the document]
Comment (Janik-Haag) [reply pkgs/applications/misc/pot/default.nix:94]: Also not quite sure.

**Comment by linsui** [comment]
Now tauri cli is used instead.

File: pkgs/applications/misc/pot/default.nix [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    cargo tauri build -b deb
+  '';
+
+  preInstall = ''
+    mv target/release/bundle/deb/*/data/usr/ $out
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix LD_LIBRARY_PATH ':' ${lib.makeLibraryPath [ libayatana-appindicator ]}
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/pot/default.nix:112]: Can we avoid using LD_LIBRARY_PATH?

Comment (linsui) [reply pkgs/applications/misc/pot/default.nix]: Fixed.

File: pkgs/applications/misc/pot/default.nix [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      rm -rf $out/v3/tmp
+      for f in $(find $out -name "*.json"); do
+        sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
+        jq --sort-keys . $f | sponge $f
+      done
+    '';
+
+    dontFixup = true;
+    outputHashMode = "recursive";
+    outputHash = "sha256-GHRgqsTwINEIqhnmcC/TaDrUDSuHXhl66kRy3T/XlT0=";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/pot/default.nix:69]: That hash seems to depend on the platform. For aarch64-linux we get ``sha256-O3ynJScipCw1ORgmC1QgM07kkqA3svv9dYvlaqpoBNk=``.

**Comment by linsui** [comment]
@Ofborg build pot

File: pkgs/applications/misc/pot/default.nix (hunk 3 above)
Comment (linsui) [reply pkgs/applications/misc/pot/default.nix]: Looks like this is fixed.

**Pull request merged**